### PR TITLE
Flanger and phaser GPU fix

### LINKED ISSUE #684: flanger incompatible with GPU
flanger torchscript test is failing on CUDA master branch.

See: https://app.circleci.com/pipelines/github/pytorch/audio?branch=master
Device mismatch happening at: https://github.com/pytorch/audio/pull/651/files#diff-292a2ef33b3ecf4c40cda1a4afe58760R1486

cc @bhargavkathivarapu

### PR DESCRIPTION
**Pull request by bhargavkathivarapu**
Fixes #684 

Issue due to   ```_generate_wave_table``` helper  function . Made it device aware .

Refactored both ```flanger``` and ```phaser``` functions that use this helper function.

### CODE REVIEW

File: torchaudio/functional.py [hunk 1]
```diff
@@ -1497,13 +1496,14 @@ def flanger(
...
-                               phase=3 * math.pi / 2)
+                               phase=3 * math.pi / 2,
+                               device=device)
 
     output_waveform = torch.zeros_like(waveform, dtype=dtype, device=device)
 
     delay_buf_pos = 0
     lfo_pos = 0
-    channel_idxs = torch.arange(0, n_channels)
+    channel_idxs = torch.arange(0, n_channels).to(device=device)
```
Comment (mthrok) [inline torchaudio/functional.py:1506]: Same here.

File: torchaudio/functional.py [hunk 2]
```diff
@@ -1317,7 +1317,6 @@ def phaser(
     delay_buf = torch.zeros(waveform.shape[0], delay_buf_len, dtype=dtype, device=device)
 
     mod_buf_len = int(sample_rate / mod_speed + .5)
-    mod_buf = torch.zeros(mod_buf_len, dtype=dtype, device=device)
```
Comment (mthrok) [inline torchaudio/functional.py:1320]: Is this not used?

File: torchaudio/functional.py [hunk 3]
```diff
@@ -1364,18 +1365,18 @@ def _generate_wave_table(
...
-
+        device (torch.device): Torch device on which table must be generated
     Returns:
         Tensor: A 1D tensor with wave table values
     """
 
     phase_offset = int(phase / math.pi / 2 * table_size + 0.5)
 
-    t = torch.arange(table_size).to(torch.int32)
+    t = torch.arange(table_size, device=device).to(torch.int32)
```
Comment (mthrok) [inline torchaudio/functional.py:1375]: You can directly path both `drive` and `dtype` arguments to `torch.arange`.

https://pytorch.org/docs/master/generated/torch.arange.html#torch-arange

File: torchaudio/functional.py [hunk 4]
```diff
@@ -1364,18 +1365,18 @@ def _generate_wave_table(
...
 
     point = (t + phase_offset) % table_size
 
-    d = torch.zeros_like(point).to(torch.float64)
+    d = torch.zeros_like(point, device=device).to(torch.float64)
```
Comment (mthrok) [inline torchaudio/functional.py:1379]: You can directly path both `drive` and `dtype` arguments to `torch.zeros_like`.

https://pytorch.org/docs/master/generated/torch.zeros_like.html#torch-zeros-like

File: torchaudio/functional.py (hunk 2 above)
Comment (bhargavkathivarapu) [reply torchaudio/functional.py:1320]: @mthrok , ```mod_buf``` is the output of ```_generate_wave_table``` . So removed this redundant intialization.

**Comment by bhargavkathivarapu** [comment]
@mthrok , changed the code to use device and dtype inside same function call

**Comment by mthrok** [comment]
Thanks!

**Pull request merged**